### PR TITLE
AutoComplete clean-up

### PR DESCRIPTION
**Pull request by carkom**
Organizing and splitting filters Autocomplete. Splitting out headers function that was duplicated in all autocomplete elements.

### CODE REVIEW

File: packages/desktop-client/src/components/autocomplete/FilterList.tsx [hunk 1]
```diff
@@ -30,6 +25,7 @@ function FilterList<T extends { id: string; name: string }>({
           ...(!embedded && { maxHeight: 175 }),
         }}
       >
+        <Fragment>{ItemHeader({ title: 'Saved Filters' })}</Fragment>
```
Comment (twk3) [inline packages/desktop-client/src/components/autocomplete/FilterList.tsx:28]: We should pass a type here so we aren't using `undefined`

Also, if we aren't using a key, should we just use <> for a fragment?

And is a fragment required here for ItemHeader?

Comment (carkom) [reply packages/desktop-client/src/components/autocomplete/FilterList.tsx:28]: Fragment isn't needed. Fixed it.

File: packages/desktop-client/src/components/autocomplete/FilterList.tsx [hunk 2]
```diff
@@ -30,6 +25,7 @@ function FilterList<T extends { id: string; name: string }>({
           ...(!embedded && { maxHeight: 175 }),
         }}
       >
+        <ItemHeader title="Saved Filters" />
```
Comment (twk3) [inline packages/desktop-client/src/components/autocomplete/FilterList.tsx:28]: @carkom should we pass a `type` here? Currently we are getting a data-testid here of `Saved-Filters-undefined-item-group` for this component.

**Comment by twk3** [comment]
@carkom I think we just need to get the ci passing, then this is complete

**Review by twk3 — APPROVED** [review]
Looks good!

**Pull request merged**